charts/redpanda/post_install_upgrade_job.tpl.go

```diff
@@ -0,0 +1,179 @@
+			!helmette.Empty(values.Storage.Tiered.CredentialsSecretRef.SecretKey.Name) &&
+			!helmette.Empty(values.Storage.Tiered.CredentialsSecretRef.SecretKey.Key) {
+			envars = append(envars, corev1.EnvVar{
+				Name: "RPK_CLOUD_STORAGE_AZURE_SHARED_KEY",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{Name: values.Storage.Tiered.CredentialsSecretRef.SecretKey.Name},
+						Key:                  values.Storage.Tiered.CredentialsSecretRef.SecretKey.Key,
+					},
+				},
+			})
+		}
+	} else {
+		if !helmette.Empty(tieredStorageConfig["cloud_storage_secret_key"]) {
+			envars = append(envars, corev1.EnvVar{
+				Name:  "RPK_CLOUD_STORAGE_SECRET_KEY",
+				Value: tieredStorageConfig["cloud_storage_secret_key"].(string),
+			})
+		} else if values.Storage.Tiered.CredentialsSecretRef.SecretKey != nil &&
+			!helmette.Empty(values.Storage.Tiered.CredentialsSecretRef.SecretKey.Name) &&
+			!helmette.Empty(values.Storage.Tiered.CredentialsSecretRef.SecretKey.Key) {
+			envars = append(envars, corev1.EnvVar{
+				Name: "RPK_CLOUD_STORAGE_SECRET_KEY",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{Name: values.Storage.Tiered.CredentialsSecretRef.SecretKey.Name},
+						Key:                  values.Storage.Tiered.CredentialsSecretRef.SecretKey.Key,
+					},
+				},
+			})
+		}
+	}
+
+	if !helmette.Empty(tieredStorageConfig["cloud_storage_access_key"]) {
+		envars = append(envars, corev1.EnvVar{
+			Name:  "RPK_CLOUD_STORAGE_ACCESS_KEY",
+			Value: tieredStorageConfig["cloud_storage_access_key"].(string),
+		})
+	} else if values.Storage.Tiered.CredentialsSecretRef.SecretKey != nil &&
+		!helmette.Empty(values.Storage.Tiered.CredentialsSecretRef.AccessKey.Name) &&
+		!helmette.Empty(values.Storage.Tiered.CredentialsSecretRef.AccessKey.Key) {
+		envars = append(envars, corev1.EnvVar{
+			Name: "RPK_CLOUD_STORAGE_ACCESS_KEY",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{Name: values.Storage.Tiered.CredentialsSecretRef.AccessKey.Name},
+					Key:                  values.Storage.Tiered.CredentialsSecretRef.AccessKey.Key,
+				},
+			},
+		})
+	}
+
+	for k, v := range tieredStorageConfig {
+		if k == "cloud_storage_access_key" || k == "cloud_storage_secret_key" || k == "cloud_storage_azure_shared_key" {
+			continue
+		}
+
+		if v == nil || helmette.Empty(v) {
+			continue
+		}
+
+		if k == "cloud_storage_cache_size" && helmette.KindIs("string", v) {
+			envars = append(envars, corev1.EnvVar{
+				Name:  fmt.Sprintf("RPK_%s", helmette.Upper(k)),
+				Value: helmette.ToJSON(SIToBytes(v.(string))),
+			})
+			continue
+		}
+
+		envars = append(envars, corev1.EnvVar{
+			Name:  fmt.Sprintf("RPK_%s", helmette.Upper(k)),
+			Value: helmette.MustToJSON(v),
```

Comment:
I don't recall but do we need/want to set these variables even if tiered storage is _not_ enabled?

Reply:
There is check that would not create env vars if tiered storage is not enabled
https://github.com/redpanda-data/helm-charts/pull/1248/files#diff-a40abdea3cd1969938214bba0c5f59d9f28469a4f6f87edee2485ff3660f76cdR54